File: src/backends/base/Hook.py
from utils import debug, error

class Hook:
    
    def __init__(self, name, type):
        
        self.name = name

        self.value = ""
        
    def get_name(self):
        return self.name
    
    def on_get(self):
        pass
    
    def get(self):
        """
        returns current value
        """
        self.on_get()
        debug(self.name + " GET: " + str(self.value))
        return self.value
    
    def on_set(self):
        pass
    
    def set(self, value):
        """
        sets current value to value param
        """
        self.value = value
        self.on_set()
        debug(self.name + " SET: " + str(self.value))

File: src/backends/base-java/src/main/java/dent/backends/base/Hook.java
package dent.backends.base;

import dent.backends.base.Utils;

import java.util.Objects;

public class Hook {

    protected String name;
    protected String value;
    protected Utils utils;

    protected String oldLocal;
    protected String oldDB;
    public Hook(String name) {
        this.name = name;
        this.value = "";
        this.utils = new Utils();
    }

    public String getName() {
        return this.name;
    }

    public void onGet() {}

    public String get() {
        this.onGet();
        this.utils.debug(this.name + " GET: " + this.value);
        return this.value;
    }

    public String onSet(String value) {
        return value;
    }

    public void set(String value) {
        //this.value = value;
        this.utils.debug(this.name + " SET: " + value);
        this.value = this.onSet(value);
    }
}
